src/entity.c

```diff
@@ -2914,16 +2924,32 @@ void flecs_copy_ptr_w_id(
     size_t size,
     void *ptr)
 {
-    if (flecs_defer_cmd(stage)) {
-        flecs_defer_set(world, stage, EcsOpSet, entity, id, 
+    flecs_component_ptr_t dst = {0};
+    ecs_record_t *r = flecs_entities_get(world, entity);
+    ecs_assert(r != NULL, ECS_INTERNAL_ERROR, NULL);
+
+    /* lookup the component pointer if the entity exists */
+    if (r->table) {
+        dst = flecs_get_component_ptr(
+                world, r->table, ECS_RECORD_TO_ROW(r->row), id);
+    }
+
+    /* if the component doesn't exist, and we're deferred, copy the new
+     * component to a temporary component on the command queue */
+    bool deferred = flecs_defer_cmd(stage);
```

Comment:
I think I see why you moved the logic here (to avoid doing a duplicate lookup of `ecs_record_t`), but I'd like the logic for command creation/insertion to stay in `stage.c`. What you could do here instead is lookup the record here, and pass it as an argument to `flecs_defer_set`.

Reply:
I don't understand the context of this comment.  Is it in reference to the direct calls to the `flecs_cmd_new()` calls below?

More questions:
* Are you asking for a new argument to be added to `flecs_defer_set()` to pass a record?
*  There's currently no way to call `flecs_defer_set()` without copying or moving the value
    * even with `size == 0`, if there's a copy or move constructor, the size is ignored

Or is something like this more what you're looking for?
```cpp
if (deferred) {
    flecs_defer_set(world, stage, EcsOpAdd, entity, id, 0, NULL, false);
    flecs_defer_set(world, stage, EcsOpModified, entity, id, 0, NULL, false);
}
```

Reply:
From Discord (for completeness): the idea is to keep all the defer logic in `flecs_defer_set`, so that the calling code still looks like this:

```c
static
void flecs_copy_ptr_w_id(
    ecs_world_t *world,
    ecs_stage_t *stage,
    ecs_entity_t entity,
    ecs_id_t id,
    size_t size,
    void *ptr)
{
    ecs_record_t *r = flecs_entities_get(world, entity);
    if (flecs_defer_cmd(stage)) {
        flecs_defer_set(world, stage, EcsOpSet, r, entity, id, 
            flecs_utosize(size), ptr, false);
        return;
    }

    // regular undeferred behavior
```